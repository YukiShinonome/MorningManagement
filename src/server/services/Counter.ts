export default class Counter {
  name: string;
  counter: number;

  constructor() {
    this.name = "counter";
    this.counter = 0;
  }

  read(_req: any, _resource: any, _params: any, _config: any) {
    return Promise.resolve(this.counter);
  }

  update(_req: any, _resource: any, _params: any, _config: any) {
    return Promise.resolve(++this.counter);
  }
}
