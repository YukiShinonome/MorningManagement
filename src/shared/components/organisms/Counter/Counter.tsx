import React from "react";
import styled from "styled-components";

export default React.memo(function Counter(props: { counterValue: any }) {
  const { counterValue } = props;
  return <Root>access counter: {counterValue || ""}</Root>;
});

const Root = styled.div`
  text-align: center;
`;
