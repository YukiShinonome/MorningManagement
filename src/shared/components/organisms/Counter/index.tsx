import { connect } from "react-redux";
import { compose } from "recompose";
import { deferLoader } from "redux-async-loader";
import { counterSelector } from "../../../redux/modules/reducer";
import { increment } from "../../../redux/modules/counter";
import Counter from "./Counter";

export default compose<{ counterValue: number }, {}>(
  deferLoader((_props, { dispatch }) => dispatch(increment())),
  connect(state => ({
    counterValue: counterSelector(state as any).value,
  })),
)(Counter);
