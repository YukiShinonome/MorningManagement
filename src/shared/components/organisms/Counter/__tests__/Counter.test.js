import React from "react";
import renderer from "react-test-renderer";
import Counter from "../Counter";

test("Counter: render correctly", () => {
  const tree = renderer.create(<Counter counterValue={1} />).toJSON();
  expect(tree).toMatchSnapshot();
});
