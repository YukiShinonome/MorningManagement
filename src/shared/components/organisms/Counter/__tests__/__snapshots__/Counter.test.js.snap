// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter: render correctly 1`] = `
<div
  className="sc-bdVaJa khUSez"
>
  access counter: 
  1
</div>
`;
